common/transport/amqp/src/amqp.ts

```diff
@@ -374,7 +374,11 @@ export class Amqp {
             this._fsm.transition('disconnecting', null, err);
           },
           error: (context: EventContext) => {
-            this._fsm.transition('disconnecting', null, context.connection.error);
+            if (context && context.connection) {
```

Comment:
>f (context && context.connection) { [](start = 13, length = 35)

This is a small hack to cover a single case, but this may be more impactful because it looks like you make the same assumption in quite a few places.  Apparently Rhea can return errors that are not EventContext objects.  In particular, this happens if we try to dereference a null object inside of _onResponseMessage.  In that case, Rhea calls this with an Error object instead of an EventContext object.

Reply:
This is neat.  I recall being unsure with this particular error.  It's fairly special. As it is (now I see it) unassociated with any particular context.  This should be the only one (error indication).

---
In reply to: [214467425](https://github.com/anthonyvercolano/azure-iot-sdk-node/pull/2#discussion_r214467425) [](ancestors = 214467425)

Reply:
On further examination, you hit this because of a seg fault in the annotations area.  This happens to pick up that error throw.  This is somewhat handy, in that this event handler has been generally only hit when there is an actual programming error in my transport code.  So keeping this is actually kind of nice.  Your addition of the tests for the context and context.connection is just fine.

---
In reply to: [214983368](https://github.com/anthonyvercolano/azure-iot-sdk-node/pull/2#discussion_r214983368) [](ancestors = 214983368,214467425)